fix: changed the default value of nrext64 to 0 to avoid unstable feature of nrext64

Sparksssj opened the pull request:
**Issue number:**

Closes https://github.com/bottlerocket-os/bottlerocket/issues/4281

**Description of changes:**
A new PR for https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/251, see discussion there.

**Testing done:**



**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

Commit:
packages: drop "dependencies" table

`buildsys` is only capable of modeling build-time dependencies, and
only for dependencies that fall within the current workspace - not
anything that comes from a kit.

Using the "dependencies" table to track the runtime dependencies of
the generated RPMs is superfluous, and likely to be incorrect even
for packages that reside in the same workspace. Meanwhile, it cannot
work properly for runtime dependencies that come from kits, since the
metadata for those packages is not available.

Historically, listing these packages as a dependency caused them to
be built before the dependent package was used. The "release" package
in particular depended on most packages in the workspace, which meant
that they would be built even if none of the variants included them
as an explicit dependency.

Now all the packages in the workspace are expected to be dependencies
of the kit that includes them, so this workaround is not necessary.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
kernel: disable BPF preload and bpfilter helpers

The kernel build will automatically enable these helpers if it finds
that the target C compiler can build a userspace program that depends
on the C library. However, the helpers require additional libraries
such as libelf which are not guaranteed to be present.

Since we don't make use of precompiled BPF programs, disable these
config options rather than leaving it up to auto-detect.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Sparksssj committed:
modified kernel packages and consume config files

Sparksssj committed:
updated os.spec to make mkfs config files required

Sparksssj committed:
modified the init process for xfs, added -c to consume config files

Sparksssj commented:
@bcressey Is this logic correct this time?

yeazelm commented on packages/kernel-6.1/lts_6.1.conf:
```diff
@@ -0,0 +1,14 @@
+# V5 features that were the mkfs defaults when the upstream Linux 6.1 LTS
```
This will solve the issue for variants that don't use XFS for the BOTTLEROCKET-DATA partition where the only XFS usage is through the ephemeral storage calls to create the filesystem. We use [systemd-mkfs](https://github.com/bottlerocket-os/bottlerocket-core-kit/blob/2a673d64daaf19bee95d8565cbbb09d8815588cc/packages/release/prepare-local-fs.service#L15) which doesn't appear to take arguments besides the two we provide. We might need to see if there is a way to provide this configuration file to systemd-makefs or else variants using XFS for the data partition will continue to have this error.

Sparksssj replied:
Well. I'm not so sure if we can use a configuration file for systemd-makefs. But if we cannot, the situation just goes back to what we discussed initially: we may not want to just set the default value for nrext64 to 0 for all kernel versions. But this indeed need more discussions.

sam-berning committed:
twoliter: update twoliter to v0.5.1

Signed-off-by: Sam Berning <bernings@amazon.com>

arnaldo2792 commented on packages/kernel-5.10/kernel-5.10.spec:
```diff
@@ -16,6 +16,9 @@ Source100: config-bottlerocket
 Source221: modprobe@neuron.service.drop-in.conf
 
+# XFS configuration
+Source300: lts_5.10.conf
```
You should name this file `mkfs.xfs.conf` since the current name right now doesn't tell us much about what it is or what's it used for. The same for the other kernels.

arnaldo2792 commented on packages/kernel-5.10/kernel-5.10.spec:
```diff
@@ -83,6 +86,12 @@ Summary: Modules for the Linux kernel
 %{summary}.
 
+%package mkfs-confs
+Summary: mkfs confiigurations for different file systems
```
```suggestion
Summary: mkfs configurations for different file systems
```

ginglis13 committed:
advisories: add BRSA for nvidia-container-toolkit 1.17.0

Patched in the 3.2.0 release of the core kit.

Signed-off-by: Gavin Inglis <giinglis@amazon.com>

Sparksssj committed:
patch: source defaults from config file before CLI parsing

cezar-r committed:
kernel-5.10: update to 5.10.228

Rebase to Amazon Linux upstream version 5.10.228-219.884.amzn2.

Signed-off-by: Cezar Rata <ceradev@amazon.com>

cezar-r committed:
kernel-6.1: update to 6.1.115

Rebase to Amazon Linux upstream version 6.1.115-126.197.amzn2023.

Drop patch for cgroup v1 statistics in favor of the Amazon Linux one.

The z3fold module is deprecated upstream and is no longer enabled, so
remove it from the list of packaged modules.

Signed-off-by: Cezar Rata <ceradev@amazon.com>

ginglis13 committed:
twoliter: bump release version to 3.3.1

Signed-off-by: Gavin Inglis <giinglis@amazon.com>

ginglis13 committed:
changelog: add notes for 3.3.1 release

Signed-off-by: Gavin Inglis <giinglis@amazon.com>

Commit:
grub: add patches for Boot Loader Interface

Backport upstream patches from the series that added the Boot Loader
Interface module, which implements a different part of the interface:
support for LoaderInfo and LoaderDevicePartUUID.

The refactoring commits added a new grub_efi_set_variable_to_string
function, along with the ability to set a variable without the non-
volatile flag applied.

Add two patches of our own: one to implement the part of BLI that's
used by `systemd-analyze` to report time spent in the firmware and in
the bootloader; and another to make it so that grub_get_time_ms() on
i386 doesn't treat the time the TSC was calibrated as the epoch for
all timestamps, which otherwise hides the time spent between power on
and when GRUB is started.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

yeazelm commented on packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch:
```diff
@@ -0,0 +1,46 @@
+From: Sparks Song <shijiao@amazon.com>
+Date: Thu, 14 Nov 2024 00:17:10 +0000
+Subject: [PATCH] mkfs: source defaults from config file
+
```
We should have a commit message that explains what you are trying to do with a bit more explanation on why we are adding it.

yeazelm commented on packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch:
```diff
@@ -0,0 +1,46 @@
+ #include "libfrog/dahashselftest.h"
+ #include "proto.h"
+ #include <ini.h>
++#include <unistd.h>
```
I'd really like to avoid adding a dependency in this patch. Can we see if <ini.h> has a function we can use for this purpose of checking if the file exists? We should consider updating `cfgfile_parse()` or adding a new, specialized function that safely checks if the file is present and moves on if not present instead of adding our own file checking logic.

yeazelm commented on packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch:
```diff
@@ -0,0 +1,46 @@
+ 	 * printf(_("Default configuration sourced from %s\n"), dft.source);
+ 	 */
+ 
++	char *default_config_path = "/usr/share/xfs/mkfs.xfs.conf";
```
The more idiomatic way to do this would be to use `#define` to set a constant like this. It helps signal to the reader this never changes.

yeazelm commented on packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch:
```diff
@@ -0,0 +1,46 @@
+ 	 */
+ 
++	char *default_config_path = "/usr/share/xfs/mkfs.xfs.conf";
++	char *original_cfgfile = cli.cfgfile;  // Save the original value (from -c if specified)
```
This isn't the only thing overwritten by reading in a configuration file. If a user specifies other options, won't this logic smash over that too?

Sparksssj replied:
I took another look at the code and realized the user specific config file is parsed actually after this code change, which starts at line 4391: 	while ((c = getopt_long(argc, argv, "b:c:d:i:l:L:m:n:KNp:qr:s:CfV",
So in this case we don't need to save the original value at all because it's empty at the moment. We can just simply apply the default file here, and safely overwrite it with user specified config file if there is one using this line: 	cfgfile_parse(&cli);

yeazelm commented on packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch:
```diff
@@ -0,0 +1,46 @@
++	char *original_cfgfile = cli.cfgfile;  // Save the original value (from -c if specified)
++
++	// First try to load the default config if it exists
++	if (access(default_config_path, R_OK) == 0) {
```
See my above comment but I'd rather stay consistent with the codebase to use the libinih library for this check.

yeazelm commented on packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch:
```diff
@@ -0,0 +1,46 @@
++		cfgfile_parse(&cli);
++		printf(_("Default configuration sourced from %s\n"), default_config_path);
++	} else {
++		fprintf(stderr, _("Warning: Default config file %s not found or not readable\n"), default_config_path);
```
I don't think we should print this. We are adding in this logic and the way we know it is working is because it prints the other message, the absence of this line would indicate we didn't use a default file (and is the expected default behavior)

Sparksssj replied:
Then I will remove this line, and only print a warning when default file found but not readable

yeazelm commented on sources/api/apiserver/src/server/ephemeral_storage.rs:
```diff
@@ -326,6 +327,12 @@ pub fn format_device<S: AsRef<OsStr>>(device: S, format: &Filesystem) -> Result<
     mkfs.arg(RAID_DEVICE_NAME);
 
+    // Apply XFS config file if formatting with XFS
+    if matches!(format, Filesystem::Xfs) {
+        mkfs.arg("-c");
+        mkfs.arg(format!("options={}", XFS_MKFS_CONF));
```
With the change to `xfsprogs`, do we need this now?

yeazelm commented on packages/kernel-6.1/kernel-6.1.spec:
```diff
@@ -139,6 +142,12 @@ Summary: Modules for the Linux kernel on bare metal
+%package mkfs-confs
+Summary: mkfs confiigurations for different file systems
+
+%description mkfs-confs
```
nit:
```suggestion
%package xfs-mkfs-confs
```

Sparksssj replied:
Why would we change this? I see all other packages have a description there

arnaldo2792 replied:
I think it is a type, but rather the suggestion is to change the subpackage name to be `xfs`. I suggested the more generic `mkfs-confgs` since we could use the package to include more configurations for other filesystems if needed.

yeazelm replied:
The diff wasn't intended to be the description, sorry about that confusion. I see the intention is to make this package generic. I'm fine with this but then we should probably call out that in a comment that this package is intended to hold future other configuration files as well, rather than specifically grabbing the package for a particular filesystem. My mental model was that we would only include these files for supported filesystems, so we would group them for each one as we add them, but this works as well.

yeazelm commented on packages/os/os.spec:
```diff
@@ -114,6 +114,7 @@ Requires: (%{_cross_os}driverdog if %{_cross_os}variant-flavor(nvidia))
 %package -n %{_cross_os}apiserver
 Summary: Bottlerocket API server
 Requires: %{_cross_os}settings-plugins
+Requires: %{_cross_datadir}/xfs/mkfs.xfs.conf
```
I think we might make this a dependency of xfsprogs rather than in os.

ginglis13 committed:
packages: use upstream amazon linux sources

Signed-off-by: Gavin Inglis <giinglis@amazon.com>

Commit:
packages: add libjson-c

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
kernel-5.15: Add patch to fix IPv6 typo

This patch fixes issues with ip6tables commands that fail due to a typo.

This is the type of error that can come up:
exit status 2: ip6tables-restore v1.8.4 (legacy): unknown option "--xor-mark"

Signed-off-by: Matthew Yeazel <yeazelm@amazon.com>

Commit:
rdma-core: require parent package for devel

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
nvme-cli: build with libjson-c for plugins

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
nvme-cli: update to 2.11

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
libnvme: update to 1.11

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
nvme-cli: add patch for EBS stats

Signed-off-by: Ben Cressey <bcressey@amazon.com>

piyush-jena committed:
changelog: add notes for 3.3.2 release

piyush-jena committed:
twoliter: bump release version to 3.3.2

Commit:
libdbus: update to 1.15.12

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
libexpat: update to 2.6.4

Signed-off-by: Ben Cressey <bcressey@amazon.com>

ginglis13 committed:
advisories: backfill kernel-6.1 BRSAs for 3.3.1

Signed-off-by: Gavin Inglis <giinglis@amazon.com>

ginglis13 committed:
packages: normalize versioning for amazon-ecs-cni-plugins

Normalize the versioning of the amazon-ecs-cni-plugins package by using
its VERSION provided in the upstream. Add an Epoch of 1 to account for
past packages using a gitrev for Version.

Signed-off-by: Gavin Inglis <giinglis@amazon.com>

sumukhballal committed:
host-ctr: added host certs to host containers

Co-authored-by: Patrick J.P. Culp <jpculp@amazon.com>

tzneal committed:
allow overriding the max-pods file with one from your variant

Files in RPMs can't conflict, so we modify pluto to read from a normally
non-existent override file. A variant can then write this file to
override the default max-pod provided by the core-kit.

Commit:
packages: remove acpid

systemd-logind is installed by default and does the only thing acpid
is configured to do: respond to power-off events.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

KCSesh committed:
packages: update kubernetes-1.28

Signed-off-by: Kyle Sessions <kssessio@amazon.com>

KCSesh committed:
packages: update kubernetes-1.29

Signed-off-by: Kyle Sessions <kssessio@amazon.com>

KCSesh committed:
packages: update kubernetes-1.30

Signed-off-by: Kyle Sessions <kssessio@amazon.com>

KCSesh committed:
packages: update kubernetes-1.31

Signed-off-by: Kyle Sessions <kssessio@amazon.com>

arnaldo2792 committed:
kernel-6.1: prevent io_uring calls from hanging

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>

koooosh committed:
advisories: add BRSA for libexpat

Signed-off-by: Kush Upadhyay <kushupad@amazon.com>

Sparksssj committed:
twoliter: bump release version to 4.0.0

Sparksssj committed:
changelog: add notes for 4.0.0 release

koooosh committed:
changelog: fix PR links for recent releases

Signed-off-by: Kush Upadhyay <kushupad@amazon.com>

cbgbt committed:
host-ctr: fix bug preventing region extraction from ECR URIs

Sparksssj committed:
WIP: temp on gh

Sparksssj committed:
Merge branch 'bottlerocket-os:develop' into xfs-config-update
